processing/src/main/java/org/apache/druid/query/filter/RangeFilter.java

```diff
@@ -514,10 +514,12 @@ private Supplier<DruidLongPredicate> makeLongPredicateSupplier()
       if (hasLowerBound()) {
         ExprEval<?> lowerCast = lowerEval.castTo(ExpressionType.LONG);
         if (lowerCast.isNumericNull()) {
+          // lower value is not null, but isn't convertible to a long, so is effectively null, we treat this as not
+          // having a lower bound to be consistent with the bound filter, but perhaps this is controversial...
```

Comment:
Hmm. It seems like, to aid in SQL compatibility, it would be better for this case to be always-false. (Or, in tri-valued logic, always-`UNKNOWN`). Lower value that is not convertible to the target type is like having a comparison `x > NULL`, which is `NULL` (a.k.a. `UNKNOWN`) itself in tri-valued logic.

Reply:
yeah, that seems reasonable to change to me

Reply:
ah so I think this actually shouldn't be able to happen, because we only use the long and double predicates if the match value type is numeric, which basically means they must not be numeric null if also the check for the bounds being set at all is true.

I realized this while writing some tests using string bounds on a long column, which pushes the comparison to use the string predicate, because the match value type is stringy and so it does lexicographical comparison instead of numeric comparison. I left the tests because they seemed cool to have :shrug: